org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/MavenRepositoryCache.java

```diff
@@ -0,0 +1,88 @@
+package org.eclipse.lsp4xml.extensions.maven;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.maven.model.Model;
+import org.apache.maven.project.MavenProject;
+import org.apache.maven.settings.Settings;
+import org.apache.maven.settings.io.xpp3.SettingsXpp3Reader;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import org.eclipse.aether.repository.RemoteRepository;
+
+public class MavenRepositoryCache {
+	HashSet<RemoteRepository> repos = new HashSet<>();
+
+	private static final MavenRepositoryCache INSTANCE = new MavenRepositoryCache();
+
+	private MavenRepositoryCache() {
+		repos.add(new RemoteRepository.Builder("central", "default", "https://repo1.maven.org/maven2/").build()); // Default
+																													// maven
+																													// repository
+	}
+
+	// TODO: get user settings
```

Comment:
Getting repositories from the user & global settings should probably be for a later patch. Default/global setting location seems to vary depending on the environment and the user's settings file location should probably be configurable (Maybe through an environment variable?)